File: Angular Client/models/prescription.ts
import {User, UserEntity} from './user';
import {FormGroup} from '@angular/forms';

export interface Prescription {
  prescriptionId: number;
  visitId: number;
  drugName: string;
  drugFrequency: string;
  drugDosage: string;
  drugPeriod: string;
  drugComment: string;
  user: User | null;
}

export class PrescriptionEntity implements Prescription{
  constructor() {
    this.prescriptionId = 0;
    this.visitId = 0;
    this.drugName = '';
    this.drugFrequency = '';
    this.drugDosage = '';
    this.drugPeriod = '';
    this.drugComment = '';
    this.user = new UserEntity();
  }

  prescriptionId: number;
  visitId: number;
  drugName: string;
  drugFrequency: string;
  drugDosage: string;
  drugPeriod: string;
  drugComment: string;
  user: User | null;

  setData(form: FormGroup): void{
    this.visitId = form.get('visitId')?.value;
    this.drugName = form.get('drugName')?.value.drugName;
    this.drugDosage = form.get('drugDosage')?.value;
    this.drugFrequency = form.get('drugFrequency')?.value.drugFrequency;
    this.drugPeriod = form.get('drugPeriod')?.value.drugPeriod;
    this.drugComment = form.get('drugComment')?.value;
  }
}

File: ASP.NET Core Server/Petadmin.Core/Models/Prescription.cs
using PetAdmin.Core.Interfaces;

namespace Petadmin.Core.Models
{
    public class Prescription : IGenericDbEntity
    {
        /// <summary>
        /// Keeps the database id of the prescription.
        /// </summary>
        public int PrescriptionId { get; set; }

        /// <summary>
        /// Keeps the database visit id of the prescription.
        /// </summary>
        public int VisitId { get; set; }

        /// <summary>
        /// Gets or sets the drug name of the prescription.
        /// </summary>
        public string DrugName { get; set; }

        /// <summary>
        /// Gets or sets the drug frequency of the prescription.
        /// </summary>
        public string DrugFrequency { get; set; }

        /// <summary>
        /// Gets or sets the drug dosage of the prescription.
        /// </summary>
        public string DrugDosage { get; set; }

        /// <summary>
        /// Gets or sets the drug period of the prescription.
        /// </summary>
        public string DrugPeriod { get; set; }

        /// <summary>
        /// Gets or sets the drug comment of the prescription.
        /// </summary>
        public string DrugComment { get; set; }
    }
}
